### DISCUSSION TITLE
Removing a file with `rm -t`, `rm --trash`

### DISCUSSION THREAD

**Discussion by ghost** [nushell/nushell#13680] [Q&A] [answered]
What happens when we delete a file using `rm -t` in an environment without a "Trash" directory? Where does the file go? Will it simply be deleted if I don't have a directory to receive the file?

An example:
![image](https://github.com/user-attachments/assets/955bddb2-2568-446e-9914-a27831caa562)

**Comment by kaathewisegit** [comment]
It should throw an error, see [this piece of code][code].  If the file gets deleted silently and permanently, there's an error somewhere, and it should be filed as a bug.


[code]: https://github.com/nushell/nushell/blob/95b78eee25a876f425b22ecb05686896c7828a36/crates/nu-command/src/filesystem/rm.rs#L172-L195

**Reply by ghost** [reply]
Perfect. Maybe it's moving the file to some place that I don't know? The `TRASH_SUPPORTED` constant in the code seems to be `true` in my environment. How can I see if it's right? Do you know?

**Reply by kaathewisegit** [reply]
What's your `version | transpose key value | to md --pretty` data?

**Reply by ghost** [reply]
```
| key                | value                                                             |
| ------------------ | ----------------------------------------------------------------- |
| version            | 0.93.0                                                            |
| major              | 0                                                                 |
| minor              | 93                                                                |
| patch              | 0                                                                 |
| branch             |                                                                   |
| commit_hash        |                                                                   |
| build_os           | linux-x86_64                                                      |
| build_target       | x86_64-unknown-linux-gnu                                          |
| rust_version       | rustc 1.77.2 (25ef9e3d8 2024-04-09) (built from a source tarball) |
| cargo_version      | cargo 1.77.1                                                      |
| build_time         | 1980-01-01 00:00:00 +00:00                                        |
| build_rust_channel | release                                                           |
| allocator          | mimalloc                                                          |
| features           | default, sqlite, system-clipboard, trash, which                   |
| installed_plugins  |                                                                   |
```

Nice. It seems that I have the "trash" feature. But dunno where the file goes anyway. 😕

**Reply by kaathewisegit** [reply]
Yep, I have no idea.  You should file it as a bug in issues

**Reply by ghost** [reply, marked as answer]
Nope! It wasn't a bug! Recently I was browsing through some files and found the `tfile` I used as an example in my image by accident.

It was located in the `$env.HOME/.local/share/Trash` directory. @KAAtheWiseGit

**Reply by ghost** [reply]
I'm sorry if that seemed stupid. I don't really know how it works 😅. Thank you for your support.

**Reply by kaathewisegit** [reply]
Glad it worked out!
